### PR TITLE
test: add nan/inf clamp regression test for fused_topk_bias

### LINKED ISSUE #40457: Add fused_topk to CI. Follow-up to 39391
#39391 was merged because want it to be fixed in v0.20.

This comment left unfixed https://github.com/vllm-project/vllm/pull/39391#pullrequestreview-4144377845 :

 - add to CI
 - add `fused_topk_bias`

### PR DESCRIPTION
**Pull request by jhaotingc**
Add test_fused_topk_bias_nan_inf_clamp to cover the same NaN/Inf scenario as test_fused_topk_nan_inf_clamp but for the fused_topk_bias entry point. On CUDA, fused_topk_bias routes through the same topk_softmax/topk_sigmoid kernels fixed in #39391 (lines 131/154 of topk_softmax_kernels.cu), so the clamp already applies.

Closes #40457

## Purpose

PR #39391 fixed NaN/Inf gating logits producing duplicate expert IDs in `fused_topk` by clamping scores to 0 in `topk_softmax_kernels.cu` (lines 131/154 for the warp-level path, line 457 for the fallback path). However,
the regression test added in that PR only covered `fused_topk`. This PR adds the same nan/inf regression test for `fused_topk_bias`, which is used by DeepSeek-style models with `e_score_correction_bias`.

On CUDA, `fused_topk_bias` routes through the same `topk_softmax` / `topk_sigmoid` C++ kernels that were patched in #39391, so the clamp already applies. This PR confirms that coverage with an explicit test.

## Test Plan

- Added `test_fused_topk_bias_nan_inf_clamp` to
  `tests/kernels/moe/test_fused_topk.py`, parametrized over:
  - `dtype`: bfloat16, float16, float32
  - `scoring_func`: softmax, sigmoid
  - `bad_value`: NaN, Inf
  - `num_experts`: 6, 8, 16
  - `topk`: 3, 4
  - Total: 144 test cases
- Verified the test is automatically collected by the existing
  **Kernels MoE Test** CI step (no `.buildkite/` changes needed):
  ```bash
  # Simulate exactly what buildkite runs:
  pytest tests/kernels/moe --collect-only -q | grep nan_inf
  # Shows both test_fused_topk_nan_inf_clamp (added in #39391) and
  # test_fused_topk_bias_nan_inf_clamp (this PR) are collected
  ```
- Ran the full `test_fused_topk.py` suite (720 tests) to check for regressions.

### nan/inf regression tests (144 cases)

| Kernel clamp | nan/inf tests passed | nan/inf tests failed |
|---|---|---|
| Disabled (pre-#39391 behaviour) | 36 | **108** |
| Enabled (post-#39391, this PR) | **144** | 0 |

### Full test suite (720 cases, 2×H200, CUDA 13.0)

```
720 passed, 16 warnings in 104.91s
```
---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [x] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [x] The test plan, such as providing test command.
- [x] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
</details>

### CODE REVIEW

**Comment by jhaotingc** [comment]
@claude review

**Comment by jhaotingc** [comment]
cc @vadiklyutiy

**Comment by vadiklyutiy** [comment]
I think we also should add this test to CI. I don't see the tests name or corresponding dir in `.buildkite`

**Comment by jhaotingc** [comment]
> I think we also should add this test to CI. I don't see the tests name or corresponding dir in `.buildkite`

Verified the test is automatically collected by the existing Kernels MoE Test CI step (no .buildkite/ changes needed):
```
# Simulate exactly what buildkite runs:
pytest tests/kernels/moe --collect-only -q | grep nan_inf
# Shows both test_fused_topk_nan_inf_clamp (added in #39391) and
# test_fused_topk_bias_nan_inf_clamp (this PR) are collected
```

Does this suffice? I'm not familiar with VLLM CI. Thanks.

**Comment by jhaotingc** [comment]
https://github.com/vllm-project/vllm/blob/main/.buildkite/test_areas/kernels.yaml#L66

**Comment by vadiklyutiy** [comment]
> Does this suffice? I'm not familiar with VLLM CI. Thanks.

Oops, missed it. All good.

**Review by vadiklyutiy — APPROVED** [review]
